src/tests/multihost/alltests/test_misc_proxy.py

```diff
@@ -0,0 +1,222 @@
+        # Auth should work using ldap provider
+        ssh1 = SSHClient(client_e, username="foo1", password="Secret123")
+        ssh1.close()
+        execute_cmd(multihost, "chown -R foo1 ~foo1")
+        execute_cmd(multihost, "chown -R foo1 /var/spool/mail/foo1")
+        # error logged in log
+        execute_cmd(multihost, "systemctl stop sssd.service")
+        execute_cmd(multihost, "rm -rf /var/lib/sss/{db,mc}/*")
+        execute_cmd(multihost, "systemctl start sssd.service")
+        execute_cmd(multihost, 'su - foo1 -c "logout"')
+
+    def test_bz785902(self, multihost):
+        """
+        :title: Errors with empty loginShell and proxy provider bz785902
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=785902
+        :id: 17831b7a-b252-11ec-9942-845cf3eff344
+        :steps:
+          1. Adding the user in ldap server with empty login shell
+          2. Search for Internal Error
+          3. Non existing netgroup returned with proxy
+          provider when proxy lib name is ldap
+        :expectedresults:
+          1. Should succeed
+          2. Should not succeed
+          3. Should not succeed
+        """
+        # Errors with empty loginShell and proxy provider bz785902
+        tools = sssdTools(multihost.client[0])
+        master_e = multihost.master[0].ip
+        ldap_uri = f'ldap://{master_e}'
+        ds_rootdn = 'cn=Directory Manager'
+        ds_rootpw = 'Secret123'
+        ldap_inst = LdapOperations(ldap_uri, ds_rootdn, ds_rootpw)
+        user_dn = 'uid=foo1,ou=People,dc=example,dc=test'
+        del_member = [(ldap.MOD_REPLACE, 'loginShell', "".encode('utf-8'))]
+        (ret, _) = ldap_inst.modify_ldap(user_dn, del_member)
+        assert ret == 'Success'
+        domain_name = tools.get_domain_section_name()
+        domain_params = {'proxy_pam_target': 'sssdproxyldap',
+                         'id_provider': 'proxy',
+                         'proxy_lib_name': 'ldap',
+                         'auth_provider': 'proxy'}
+        tools.sssd_conf('domain/' + domain_name, domain_params)
+        tools.clear_sssd_cache()
+        execute_cmd(multihost, "id foo1")
+        for error_error in ['[sysdb_set_entry_attr] (6): '
+                            'Error: 14 (Bad address)',
+                            '[sysdb_store_user] (6): '
+                            'Error: 14 (Bad address)',
+                            'Internal Error (Cannot make/remove '
+                            'an entry for the specified session)']:
+            with pytest.raises(subprocess.CalledProcessError):
+                execute_cmd(multihost, f"grep {error_error} /var/log/sssd/*")
+        del_member = [(ldap.MOD_REPLACE, 'loginShell',
+                       "/bin/bash".encode('utf-8'))]
+        (ret, _) = ldap_inst.modify_ldap(user_dn, del_member)
+        assert ret == 'Success'
+        # non existing netgroup returned with proxy provider
+        # when proxy lib name is ldap bz804103
```

Comment:
This is a tests for a different BZ and should have a test in its own.

Reply:
Done